holidays/countries/papua_new_guinea.py

```diff
@@ -0,0 +1,185 @@
+    shall, unless the Head of State, acting on advice, declares otherwise, be observed as a
+    public holiday throughout the country.
+
+    When Christmas Day falls on a Sunday, the following Tuesday shall also be observed as a
+    public holiday.
+    """
+
+    country = "PG"
+    observed_label = "%s (Observed)"
+
+    @property
+    def sovereign_birthday(self) -> str:
+        """Sovereign's birthday holiday name."""
+        return (
+            # King's Birthday.
+            "King's Birthday"
+            if self._year >= 2023
+            # Queen's Birthday.
+            else "Queen's Birthday"
+        )
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        StaticHolidays.__init__(self, PapuaNewGuineaStaticHolidays)
+        kwargs.setdefault("observed_rule", SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # Public Holidays Law 1953 (No. 38 of 1953).
+        if self._year <= 1952:
+            return None
+
+        # Section 1: Public Holidays.
+        # - Easter Saturday is currently not gazetted as a Public Holiday in 2024.
+        # - While Easter Sunday itself is not listed, this is de facto always a day-off.
+
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day("New Year's Day"))
+
+        # Good Friday.
+        self._add_good_friday("Good Friday")
+
+        if self._year <= 2023:
+            # Easter Saturday.
+            self._add_holy_saturday("Easter Saturday")
+
+        # Easter Sunday.
+        self._add_easter_sunday("Easter Sunday")
+
+        # Easter Monday.
+        self._add_easter_monday("Easter Monday")
+
+        # Papua New Guinea Remembrance Day.
+        self._add_observed(self._add_holiday_jul_23("Papua New Guinea Remembrance Day"))
+
+        # Christmas Day.
+        self._add_observed(self._add_christmas_day("Christmas Day"), rule=SUN_TO_NEXT_TUE)
+
+        # Boxing Day.
+        self._add_observed(self._add_christmas_day_two("Boxing Day"))
+
+        # Section 2: Independence Day or Days.
+
+        # Independence Day
+        # Commemorates the attainment of Independent Sovereign Statehood on Sep 16, 1975.
+
+        if self._year >= 1975:
+            # Independence Day.
+            self._add_observed(self._add_holiday_sep_16("Independence Day"))
+
+        # Section 3: Anniversary of Birthday of the Queen/King.
+        # In Papua New Guinea, it is usually celebrated on the second Monday of June every year.
+        # From 2023, the date changed to Jun 17 (Special Observed for 2023 on Jun 16).
+        # For localization, see "self.sovereign_birthday".
+
+        if self._year <= 2022:
+            self._add_holiday_2nd_mon_of_jun(self.sovereign_birthday)
+        else:
+            self._add_holiday_jun_17(self.sovereign_birthday)
```

Comment:
```suggestion
        if self._year <= 2022:
            # Queen's Birthday.
            self._add_holiday_2nd_mon_of_jun("Queen's Birthday")
        else:
            # King's Birthday.
            self._add_holiday_jun_17("King's Birthday")
```
I think special property is a bit excessive here.

Reply:
Fixed.